crates/sage/src/scoring.rs

```diff
@@ -522,7 +545,7 @@ impl<'db> Scorer<'db> {
     }
 
     /// Calculate full hyperscore for a given PSM
-    fn score_candidate(&self, query: &ProcessedSpectrum, pre_score: &PreScore) -> Score {
+    fn score_candidate(&self, query: &ProcessedSpectrum, pre_score: &PreScore) -> (Score,Fragments) {
```

Comment:
Surely we *only* want to actually keep track of annotated fragments if the user has specifically requested it. Currently it appears that PSMs are being annotated even if the user did not ask for this to happen, and I imagine that all of the memory allocations here will be very expensive

Reply:
An additional condition has been included to account for the lack of annotated fragments. If the user does not specify the parameter, the system will not track the fragments.